pkg/apis/hive/v1/validating-webhooks/clusterdeployment_validating_admission_hook.go

```diff
@@ -421,9 +422,24 @@ func (a *ClusterDeploymentValidatingAdmissionHook) validateUpdate(admissionSpec
 		}
 	}
 
-	if oldObject.Spec.Installed && !newObject.Spec.Installed {
-		allErrs := field.ErrorList{}
-		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "installed"), newObject.Spec.Installed, "cannot make uninstalled once installed"))
+	allErrs := field.ErrorList{}
+	specPath := field.NewPath("spec")
+
+	if newObject.Spec.Installed {
+		if newObject.Spec.ClusterMetadata != nil {
+			if oldObject.Spec.Installed {
```

Comment:
This might be worth a comment, I stared at this for a couple minutes thinking something was wrong, it wasn't, but it's a little unclear that this is ensuring clustermetadata is immutable once set.

Reply:
A comment sounds like a great idea.